trie_test.go

```diff
@@ -87,48 +87,17 @@ func TestMatchNilTrie(t *testing.T) {
 	assert.Nil(t, trie.Match(key))
 }
 
-func TestInsertNodeWithChildren(t *testing.T) {
```

Comment:
Since the caller does not create and insert nodes directly, this test is no longer relevant. I consider this a good sign that this API change is the right thing to do.